File: Client/Android/solutions/VideoPlaybackEdit/Vod/vod-scenekit/src/main/java/com/bytedance/vod/scenekit/utils/ViewUtils.java
// Copyright (c) 2023 BytePlus Pte. Ltd.
// SPDX-License-Identifier: Apache-2.0

package com.bytedance.vod.scenekit.utils;

import android.content.res.Resources;
import android.view.View;
import android.view.ViewGroup;

public class ViewUtils {

    public static ViewGroup removeFromParent(View view) {
        ViewGroup parent = (ViewGroup) view.getParent();
        if (parent != null) {
            parent.removeView(view);
        }
        return parent;
    }

    public static int dp2px(float dpValue) {
        final float scale = Resources.getSystem().getDisplayMetrics().density;
        return (int) (dpValue * scale + 0.5f);
    }
}

File: Client/Android/solutions/vod-live-feed/src/main/java/com/byteplus/vodlive/utils/ViewUtils.kt
// Copyright (c) 2023 BytePlus Pte. Ltd.
// SPDX-License-Identifier: Apache-2.0

package com.byteplus.vodlive.utils

import android.content.Context

object ViewUtils {
    fun dip2Px(context: Context, dipValue: Float): Float {
        val scale: Float = context.resources.displayMetrics.density
        return dipValue * scale + 0.5F
    }
}